form/form.go

```diff
@@ -0,0 +1,354 @@
+	t := val.Type()
+
+	// Also do nothing if this is the type's zero value
+	if t.Comparable() && val.Interface() == reflect.Zero(t).Interface() {
+		return
+	}
+
+	// Special case for when a type has implemented special logic to append
+	// itself to a form.
+	if t.Implements(reflect.TypeOf((*Appender)(nil)).Elem()) {
+		val.Interface().(Appender).AppendTo(values, keyParts)
+	}
+
+	switch val.Kind() {
+	case reflect.Array, reflect.Slice:
+		// FormatKey automatically adds square brackets, so just pass an empty
+		// string into the breadcrumb trail
+		arrNames := append(keyParts, "")
+
+		for i := 0; i < val.Len(); i++ {
+			// The one exception to the above is when options have requested
+			// that this array/slice be indexed. In that case we produce a hash
+			// keyed with integers which the Stripe API knows how to interpret.
+			if options != nil && options.IndexedArray {
+				arrNames = append(keyParts, strconv.Itoa(i))
+			}
+
+			reflectValue(values, val.Index(i), arrNames, nil)
+		}
+
+	case reflect.Bool:
+		if options != nil {
+			if val.Bool() {
+				switch {
+				case options.Empty:
+					values.Add(FormatKey(keyParts), "")
+				case options.Invert:
+					values.Add(FormatKey(keyParts), strconv.FormatBool(false))
+				case options.Zero:
+					values.Add(FormatKey(keyParts), "0")
+				}
+			}
+		} else {
+			values.Add(FormatKey(keyParts), strconv.FormatBool(val.Bool()))
+		}
+
+	case reflect.Float32:
+		values.Add(FormatKey(keyParts), strconv.FormatFloat(val.Float(), 'f', 4, 32))
+
+	case reflect.Float64:
+		values.Add(FormatKey(keyParts), strconv.FormatFloat(val.Float(), 'f', 4, 64))
+
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64:
+		values.Add(FormatKey(keyParts), strconv.FormatInt(val.Int(), 10))
+
+	case reflect.Interface:
+		if val.IsNil() {
+			return
+		}
+		reflectValue(values, val.Elem(), keyParts, nil)
+
+	case reflect.Map:
+		for _, keyVal := range val.MapKeys() {
+			if Strict && keyVal.Kind() != reflect.String {
+				panic("Don't support serializing maps with non-string keys")
+			}
+
+			reflectValue(values, val.MapIndex(keyVal), append(keyParts, keyVal.String()), nil)
+		}
+
+	case reflect.Ptr:
+		if val.IsNil() {
+			return
+		}
+		reflectValue(values, val.Elem(), keyParts, options)
+
+	case reflect.String:
+		values.Add(FormatKey(keyParts), val.String())
+
+	case reflect.Struct:
+		for i := 0; i < t.NumField(); i++ {
+			field := t.Field(i)
+			tag := field.Tag.Get(tagName)
+			if Strict && tag == "" {
+				panic(fmt.Sprintf(
+					"All fields in structs to be form-encoded must have `form` tag; on: %s/%s "+
+						"(hint: use an explicit `form:\"-\"` if the field should not be encoded",
+					t.Name(), field.Name,
+				))
+			}
+
+			formName, options := parseTag(tag)
+
+			// Like with encoding/json, a hyphen is an explicit way of saying
+			// that this field should not be encoded
+			if formName == "-" {
+				continue
+			}
+
+			if Strict && options != nil &&
```

Comment:
Yeah. I added it in though just in case there's some params struct somewhere that I missed invoking in any test and which might then `panic` someone's production app. I was on the fence about adding this though, and I think given a few weeks of people using this version and no bugs reported we can just remove it.